### PR TITLE
[Analysis] Allow ignoring leading, trailing and consecutive zeroes in…

### PR DESCRIPTION
**Pull request by rbayet**
… reference fields

### CODE REVIEW

File: src/module-elasticsuite-core/Helper/IndexSettings.php [hunk 1]
```diff
@@ -40,6 +41,28 @@ class IndexSettings extends AbstractConfiguration
...
+
+    /**
+     * @var string
+     */
+    const REFERENCE_ANALYZER_CONFIG_XML_PREFIX = 'reference_analyzer';
+
+    /**
+     * @var string
+     */
+    const OLD_DEFAULT_INDICES_PATTERN = '{{YYYYMMdd}}_{{HHmmss}}';
```
Comment (romainruaud) [inline src/module-elasticsuite-core/Helper/IndexSettings.php:59]: do we need this in this commit ?

File: src/module-elasticsuite-core/Helper/IndexSettings.php [hunk 2]
```diff
@@ -40,6 +41,28 @@ class IndexSettings extends AbstractConfiguration
...
+
+    /**
+     * @var string
+     */
+    const DEFAULT_INDICES_PATTERN = '{{Ymd}}_{{His}}';
```
Comment (romainruaud) [inline src/module-elasticsuite-core/Helper/IndexSettings.php:64]: same as above

File: src/module-elasticsuite-core/Helper/IndexSettings.php (hunk 1 above)
Comment (rbayet) [reply src/module-elasticsuite-core/Helper/IndexSettings.php:59]: Strange, this is something related to the cherry-pick from the based-on-2.11.x feature branch that I thought was removed.

File: src/module-elasticsuite-core/Helper/IndexSettings.php (hunk 2 above)
Comment (rbayet) [reply src/module-elasticsuite-core/Helper/IndexSettings.php:64]: Same as above

**Pull request merged**